bootstrap/pkg/kfapp/kustomize/kustomize.go

```diff
@@ -286,189 +259,73 @@ func (kustomize *kustomize) Apply(resources kftypesv3.ResourceEnum) error {
 				Message: fmt.Sprintf("error evaluating kustomization manifest for %v Error %v", app.Name, err),
 			}
 		}
+		//TODO this should be streamed
 		data, err := resMap.EncodeAsYaml()
 		if err != nil {
 			return &kfapisv3.KfError{
 				Code:    int(kfapisv3.INTERNAL_ERROR),
 				Message: fmt.Sprintf("can not encode component %v as yaml Error %v", app.Name, err),
 			}
 		}
-		resourcesErr := kustomize.deployResources(kustomize.restConfig, data)
-		if resourcesErr != nil {
-			return &kfapisv3.KfError{
-				Code:    int(kfapisv3.INTERNAL_ERROR),
-				Message: fmt.Sprintf("couldn't create resources from %v Error: %v", app.Name, resourcesErr),
-			}
+		err = apply.Apply(data)
+		if err != nil {
+			return err
```

Comment:
Currently when error occurs, we stop for loop and return error.
Can we instead 
1. still finish the for loop while pile up errors
2. return `[]error` when `apply` finished?

I feel 1 or 2 failed resource shouldn't block applying the rest?

Reply:
Hi @kunmingg 

kubectl's apply.run will aggregate errors and continue with best effort.
I added cmdutil.checkErr(resourcesErr) if resourcesErr != nil.
CheckErr will print out multiple errors if it has aggregated more than one.